File: covid-api/src/main/java/com/tacs/grupo1/covid/api/dto/Country.java
package com.tacs.grupo1.covid.api.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;
import java.time.OffsetDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
public class Country {
    private Long id;
    private String name;
    @JsonProperty("iso_country_code")
    private String isoCountryCode;
    @JsonProperty("start_date")
    private OffsetDateTime startDate;
    @JsonProperty("offset")
    private Long offsetVal;
    @JsonProperty("strategy")
    private String strategy;

}

File: covid-frontend/src/models/Country.ts
export type CountryData = {
    id: number;
    name: string;
    iso_country_code: string;
    start_date: Date;
    offset: number;
    strategy:string;
}

export default class Country {
    id: number;
    name: string;
    iso_country_code: string;
    start_date: Date;
    offset: number;
    strategy:string;

    constructor(country?: CountryData){
        if(country){
            this.id = country.id;
            this.name = country.name;
            this.iso_country_code = country.iso_country_code;
            this.start_date = country.start_date;
            this.offset = country.offset;
            this.strategy = country.strategy;
        }else{
            this.id = 0;
            this.name = "";
            this.iso_country_code = "";
            this.start_date = new Date();
            this.offset = 0;
            this.strategy = "";
        }
    }
}